### PR TITLE
fix: allow handling floating values

### PR DESCRIPTION
**Pull request by josoriom**

### CODE REVIEW

File: src/draw/drawMarker.ts
Comment (targos) [inline src/draw/drawMarker.ts:1]: I'm not sure we should round in `drawMarker`. We can let the functions called internally handle float values.

File: src/draw/drawRectangle.ts [hunk 1]
```diff
@@ -56,14 +56,16 @@ export function drawRectangle(
...
     origin = { column: 0, row: 0 },
-    width = image.width,
-    height = image.height,
     strokeColor = getDefaultColor(image),
     fillColor = 'none',
   } = options;
-  const { column, row } = origin;
-
+  const width = Math.round(rectangleWidth ?? image.width);
+  const height = Math.round(rectangleHeight ?? image.width);
```
Comment (targos) [inline src/draw/drawRectangle.ts:66]:
```suggestion
  const width = Math.round(rectangleWidth);
  const height = Math.round(rectangleHeight);
```

You already set the default value above.

File: src/draw/drawMarker.ts [hunk 2]
```diff
@@ -89,8 +89,8 @@ export function drawMarker(
   }
   if (shape === 'square') {
     const origin = {
-      row: Math.floor(point.row - size / 2),
-      column: Math.floor(point.column - size / 2),
+      row: Math.round(point.row - size / 2),
+      column: Math.round(point.column - size / 2),
```
Comment (targos) [inline src/draw/drawMarker.ts:93]:
```suggestion
      row: point.row - size / 2,
      column: point.column - size / 2,
```

**Review by targos — CHANGES_REQUESTED** [review]

File: src/draw/drawMarker.ts [hunk 3]
```diff
@@ -89,8 +89,8 @@ export function drawMarker(
   }
   if (shape === 'square') {
     const origin = {
-      row: Math.floor(point.row - size / 2),
-      column: Math.floor(point.column - size / 2),
+      row: Math.round(point.row - size / 2),
```
Comment (lpatiny) [inline src/draw/drawMarker.ts:92]: I would use size-1 for row and col, seems to make more sense to me
`row: Math.round(point.row - (size-1) / 2)`
You want that if the size is 1 the pixel is exactly at the specified position.

File: src/draw/drawRectangle.ts [hunk 4]
```diff
@@ -56,14 +56,16 @@ export function drawRectangle(
...
+    height: rectangleHeight = image.height,
     origin = { column: 0, row: 0 },
-    width = image.width,
-    height = image.height,
     strokeColor = getDefaultColor(image),
     fillColor = 'none',
   } = options;
-  const { column, row } = origin;
-
+  const width = Math.round(rectangleWidth ?? image.width);
```
Comment (lpatiny) [inline src/draw/drawRectangle.ts:65]: rectangleWidth should always be defined and set to the default value image.width so `?? image.width` seems useless.

**Review by lpatiny — CHANGES_REQUESTED** [review]

File: src/draw/drawRectangle.ts (hunk 1 above)
Comment (josoriom) [reply src/draw/drawRectangle.ts:66]: You're right, I did it twice

**Pull request merged**